shopelectro/views/catalog.py

```diff
@@ -38,19 +39,27 @@ class CategoryTree(catalog.CategoryTree):
     category_model = models.Category
 
 
-def prepare_tile_products(products):
-    images = Image.objects.get_main_images_by_pages(
-        models.ProductPage.objects.filter(
-            shopelectro_product__in=products
+class ProductsAdapter:
+    """Adapts products QuerySet for using in template."""
+
+    def __init__(self, products: QuerySet):
+        self.products = products
+
+    # @todo #388:15m Annotate method `ProductsAdapter.prepare`.
+    #  With namedtuple, dataclass or smth.
+    def prepare(self):
```

Comment:
`TileProducts(...).get()` also would sound better

Reply:
@artemiy312 , it's confusing to see QuerySet as output of get method.
I'll suggest some differ approach

Reply:
@duker33 Result of the method is list of tuples